### ISSUE TITLE
clean up exception when lazy loading attempted

### ISSUE THREAD

**Issue by gavinking** [hibernate/hibernate-rx#28]
Currently, an attempt to lazy load an association produces this error:

```
java.lang.ClassCastException: java.util.concurrent.CompletableFuture incompatible with org.hibernate.rx.EagerUniManyToOneAssociationTest$Book
	at org.hibernate.proxy.pojo.bytebuddy.ByteBuddyInterceptor.intercept(ByteBuddyInterceptor.java:53)
	at org.hibernate.proxy.ProxyConfiguration$InterceptorDispatcher.intercept(ProxyConfiguration.java:95)
```

We need to produce something more meaningful, telling you to use `fetch()`.

**Comment by gavinking** [comment]
So fixing this one, which I figured would be easy, turns out to be quite hard without making changes to Hibernate core. I thought I would be able to do it by replacing the `BytecodeProvider` but it turns out that only three hardcoded `BytecodeProvider`s are supported, so that ain't gunna work.

If we could override `SessionImpl.immediateLoad()` we would be good, but we can't. So it looks like we probably need to patch core. :-/

**Comment by gavinking** [comment]
So this problem and the problem in #12 would both be fixable if `RxHibernateSessionImpl` extended `SessionImpl` instead of wrapping it. Unfortunately `SessionImpl` is declared `final` in core. :-/

**Comment by gavinking** [comment]
Done, though the fix depends on #36.
